### DISCUSSION TITLE
Configure headless nvim for attaching with Neovide?

### DISCUSSION THREAD

**Discussion by lkhphuc** [neovide/neovide#2264] [Q&A] [answered]
In my nvim config, I have a block to disable terminal animation and setup keybinding, etc.:
```lua
if vim.g.neovide then
  vim.g.minianimate_disable = true
  vim.g.neovide_input_macos_alt_is_meta = true
  vim.keymap.set("v", "<D-c>", '"+y') -- Copy
  vim.keymap.set({ "n", "v" }, "<D-v>", '"+P') -- Paste
  vim.keymap.set({ "i", "c" }, "<D-v>", "<C-R>+") -- Paste
  vim.keymap.set("t", "<D-v>", [[<C-\><C-N>"+P]]) -- Paste
...
```
Everything works fine when I start neovide directly. But if I start nvim in headless for remote attach like `nvim --headless --listen some-existing-dir/my-nvim-instance.sock`, later when I attach Neovide UI the configuration will not be executed because supposedly the config file is executed before neovide is attached.

How do I properly configure neovide in this case?

**Comment by fredizzimo** [comment, marked as answer]
You can either use environment variables instead of `vim.g.neovide`, or use
https://neovim.io/doc/user/starting.html#--cmd

`nvim --headless --listen some-existing-dir/my-nvim-instance.sock --cmd "let g:neovide=1"`
